src/Discord/Builders/Components/Container.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+/*
+ * This file is a part of the DiscordPHP project.
+ *
+ * Copyright (c) 2015-present David Cole <david.cole1340@gmail.com>
+ *
+ * This file is subject to the MIT license that is bundled
+ * with this source code in the LICENSE.md file.
+ */
+
+namespace Discord\Builders\Components;
+
+/**
+ * Containers are a new way to group components together.
+ * You can also specify an accent color (similar to embeds) and spoiler it.
+ *
+ * @link https://discord.com/developers/docs/interactions/message-components#container
+ *
+ * @since 10.4.0
+ */
+class Container extends Component implements Contracts\ComponentV2
+{
+    /**
+     * Array of components.
+     *
+     * @var Component[]
+     */
+    private $components = [];
+
+    /**
+     * Accent color for the container.
+     *
+     * @var int|null
+     */
+    private $accent_color;
+
+    /**
+     * Whether the container is a spoiler.
+     *
+     * @var bool
+     */
+    private $spoiler = false;
+
+    /**
+     * Creates a new container.
+     *
+     * @return self
+     */
+    public static function new(): self
+    {
+        return new self();
+    }
+
+    /**
+     * Adds a component to the container.
+     *
+     * @param Component $component Component to add.
+     *
+     * @throws \InvalidArgumentException Component is not a valid type.
+     * @throws \OverflowException        Container exceeds 10 components.
+     *
+     * @return $this
+     */
+    public function addComponent(Component $component): self
+    {
+        if (count($this->components) >= 10) {
+            throw new \OverflowException('You can only have 10 components per container.');
+        }
+
```

Comment:
might want to validate the component type that can be added here or perhaps strict type them

Reply:
Both PHPDoc and instanceof methods have been implemented.